### PR TITLE
refactor: rework exception handling and add custom exception

### PR DESCRIPTION
**Pull request by tcrasset**
This PR : 

- adds a custom `InvalidParameterException` that can be caught in the caller code e.g. the API
- move the exception handling in `projection.fit()` instead of `pca.fit()`, `famd.fit()` and `mca.fit()`
- remove `nf='all'` in favor of `nf=None` to select all columns

### CODE REVIEW

File: saiph/exception.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class InvalidParameterException(Exception):
+    """
+    Error encountered when an invalid parameter is passed.
+
+    Note that the error message gets propagated to the user as is.
+    Do NOT put any sensitive information as message.
```
Comment (charlax) [inline saiph/exception.py:6]: This sentence makes little sense in saiph, which is an open source library :)

File: saiph/inverse_transform.py [hunk 2]
```diff
@@ -39,10 +40,10 @@ def inverse_transform(
     n_records = len(coord)
 
     if not use_approximate_inverse and n_records < n_dimensions:
-        raise ValueError(
+        # Removed implementation details from the error as it will be propagated to the
+        # outside.
```
Comment (tas17) [inline saiph/inverse_transform.py:44]: I saw charlax's comment, this is the same here

Comment (tas17) [reply saiph/inverse_transform.py:44]: Maybe just remove the comment ?

File: saiph/projection.py [hunk 3]
```diff
@@ -5,14 +5,15 @@
...
+from saiph.exception import InvalidParameterException
 from saiph.models import Model
 from saiph.reduction import DUMMIES_PREFIX_SEP, famd, famd_sparse, mca, pca
 from saiph.reduction.utils.common import get_projected_column_names
 
 
 def fit(
     df: pd.DataFrame,
-    nf: Optional[Union[int, str]] = None,
+    nf: Optional[int] = None,
```
Comment (tas17) [inline saiph/projection.py:16]: Yeah, good point !

**Review by tas17 — APPROVED** [review]
Nice!
I'm not sure wether we want `saiph` to return `less` or `avatar` to cut sensible information...

**Pull request merged**